addon/ng2/tasks/uninstall.js

```diff
@@ -0,0 +1,83 @@
+/* jshint node: true, esversion: 6 */
+'use strict';
+
+const Promise           = require('ember-cli/lib/ext/promise');
+const Task              = require('ember-cli/lib/models/task');
+const chalk             = require('chalk');
+const path              = require('path');
+const fs                = require('fs');
+const fse               = require('fs-extra');
+const exec              = Promise.denodeify(require('child_process').exec);
+const packageJSON       = path.resolve(process.cwd(), 'package.json');
+const nodeModules       = path.resolve(process.cwd(), 'node_modules');
+const npmUninstall      = require('./npm-uninstall');
+const typingsUninstall  = require('./typings-uninstall');
+
+module.exports = Task.extend({
+  completionOKMessage: 'Successfully uninstalled.',
+  completionErrorMessage: 'Error uninstalling package.',
+
+  run: function(options) {
+    this.package = options.package;
+    this.typings = options.typings;
+
+    return this.uninstallProcedure();
+  },
+
+  uninstallProcedure: function() {
+    this.ui.startProgress(chalk.green(`Uninstalling package: ${this.package}`), chalk.green(`.`));
+
+    return npmUninstall(this.package)
+      .then(() => this.unlinkPackage(this.package))
+      .then(() => this.removeFromSystemJSConfig(this.package))
+      .then(() => this.ui.stopProgress())
+      .then(() => {
+        const typings = this.typings;
+        if (this.typings) {
+          this.ui.startProgress(chalk.green(`Uninstalling typings: ${typings}`), chalk.green(`.`));
+          return typingsUninstall(this.typings).then(() => this.ui.stopProgress())
+        }
+        else {
+          this.announceOKCompletion();
+        }
+      })
+      .then(() => this.announceOKCompletion());
+  },
+
+  unlinkPackage: function(pkg) {
+    const dir = path.resolve(process.cwd(), 'src', 'libs', pkg);
+    fse.removeSync(dir);
+
+    return Promise.resolve();
+  },
+
+  removeFromSystemJSConfig: function(pkg) {
+    const systemPath = path.resolve(process.cwd(), 'src', 'config', 'system.config.js');
+    let systemContents = fs.readFileSync(systemPath, 'utf8');
+
+    systemContents = systemContents.split('\n');
```

Comment:
See above. Please move both functions into a separate module and require that module.

Something like:

```typescript
export function loadSystemJson(systemPath: string): { [name: string]: any } {
  ...
}
export function saveSystemJson(systemPath: string, json: { [name: string]: any }) {
  ...
}```